src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.State.cs

```diff
@@ -265,16 +349,278 @@ private static bool IsValidAttributeParameterType(ITypeSymbol type)
+                CancellationToken cancellationToken)
+            {
+                var parameterToExistingMemberMap = ImmutableDictionary.CreateBuilder<string, ISymbol>();
+                var parameterToNewFieldMap = ImmutableDictionary.CreateBuilder<string, string>();
+                var parameterToNewPropertyMap = ImmutableDictionary.CreateBuilder<string, string>();
+
+                using var _ = ArrayBuilder<IParameterSymbol>.GetInstance(out var parameters);
+
+                for (var i = 0; i < parameterNames.Length; i++)
+                {
+                    // See if there's a matching field or property we can use.  First test in a case sensitive
+                    // manner, then case insensitively.
+                    if (!TryFindMatchingFieldOrProperty(
+                            arguments, attributeArguments, parameterNames, parameterTypes, i,
+                            parameterToExistingMemberMap, parameterToNewFieldMap, parameterToNewPropertyMap,
+                            caseSensitive: true, newParameterNames: out parameterNames, cancellationToken) &&
+                       !TryFindMatchingFieldOrProperty(
+                           arguments, attributeArguments, parameterNames, parameterTypes, i,
+                            parameterToExistingMemberMap, parameterToNewFieldMap, parameterToNewPropertyMap,
+                            caseSensitive: false, newParameterNames: out parameterNames, cancellationToken))
+                    {
+                        // If no matching field was found, use the fieldNamingRule to create suitable name
+                        var bestNameForParameter = parameterNames[i].BestNameForParameter;
+                        var nameBasedOnArgument = parameterNames[i].NameBasedOnArgument;
+                        parameterToNewFieldMap[bestNameForParameter] = _fieldNamingRule.NamingStyle.MakeCompliant(nameBasedOnArgument).First();
+                        parameterToNewPropertyMap[bestNameForParameter] = _propertyNamingRule.NamingStyle.MakeCompliant(nameBasedOnArgument).First();
+                    }
+
+                    parameters.Add(CodeGenerationSymbolFactory.CreateParameterSymbol(
+                        attributes: default,
+                        refKind: _service.GetRefKind(arguments[i]),
+                        isParams: false,
+                        type: parameterTypes[i],
+                        name: parameterNames[i].BestNameForParameter));
+                }
+
+                _parameters = parameters.ToImmutable();
+                _parameterToExistingMemberMap = parameterToExistingMemberMap.ToImmutable();
+                ParameterToNewFieldMap = parameterToNewFieldMap.ToImmutable();
+                ParameterToNewPropertyMap = parameterToNewPropertyMap.ToImmutable();
+            }
+
+            private bool TryFindMatchingFieldOrProperty(
+                ImmutableArray<TArgumentSyntax> arguments,
```

Comment:
You only ever use `arguments[index]` from this array. Seems like this could be simplified to just pass that in.

Reply:
oh?  i can clean that up then :)

Reply:
done!